include/circt/Dialect/FIRRTL/Passes.h

```diff
@@ -28,9 +28,27 @@ std::unique_ptr<mlir::Pass>
 createLowerFIRRTLAnnotationsPass(bool ignoreUnhandledAnnotations = false,
                                  bool ignoreClasslessAnnotations = false);
 
-std::unique_ptr<mlir::Pass>
-createLowerFIRRTLTypesPass(bool preserveAggregate = false,
-                           bool preservePublicTypes = true);
+/// Configure which aggregate values will be preserved by the LowerTypes pass.
+namespace PreserveAggregate {
+enum PreserveMode {
+  /// Don't preserve aggregate at all. This has been default behaivor and
+  /// compatible with SFC.
+  None,
+
+  /// Preserve only 1d vectors (e.g. UInt<2>[3]).
+  OneDimVec,
```

Comment:
It may be good to clarify that this is `1d vectors of ground type`. It would be good to clarify this in user-facing code, e.g., help messages.

Reply:
Thank you for the suggestion!